client/context.go

```diff
@@ -49,6 +49,7 @@ type Context struct {
 	NodeURI           string
 	FeeGranter        sdk.AccAddress
 	Viper             *viper.Viper
+	Codec			  codec.Codec
```

Comment:
Yeah, I guess we need Codec in clientCtx, because we need to marshal to protobuf binaru.

It's a bit useless to have both Codec and JSONCodec though. Maybe we can remove JSONCodec?

Reply:
cc @robert-zaremba

Reply:
I don't know why `JSONCodec` was added in the first place rather than `codec.Codec`. Maybe to only expose what was needed in the past? Let's check with @aaronc  - maybe there was additional motivation for that. I think we can remove the JSONCodec and use only Codec (maybe better to do it in other PR to not mess up here).